File: Assets/Scripts/OSC/OSCMetronomeLED.js
public var channel : int = 1;
public var restColor : Color = Color.black;
public var activeColor : Color = Color.red;
public var metronomeLEDs : MetronomeLED[];

class MetronomeLED {
	public var obj : GameObject;
	@System.NonSerialized public var mat : Material;
}

function Start () {
	
	// initialize all LEDs
	for(metronomeLED in metronomeLEDs) {
		metronomeLED.mat = metronomeLED.obj.gameObject.GetComponent(Renderer).material;
		SetMetronomeLEDColor(metronomeLED, restColor);
	}
}

function SetMetronomeLEDColor(metronomeLED : MetronomeLED, inputColor : Color) {
	metronomeLED.mat.SetColor("_Color", inputColor);
	metronomeLED.mat.SetColor("_TintColor", inputColor);
	metronomeLED.mat.SetColor("_Emission", inputColor);
	//metronomeLED.mat.SetColor("_TintColor", inputColor);
}

function OSCMetronome (tempo) {
	StopCoroutine( "StrobeLED");	
	StartCoroutine( "StrobeLED", 0.5);
}

function StrobeLED(fadeTime) {
	var startTime = Time.time;
	var normalizedEelapsedTime = 0.0;
	
	while(normalizedEelapsedTime <=1.0){
		elapsedTime = (Time.time - startTime);
		normalizedEelapsedTime = elapsedTime/fadeTime;
		lerpedColor = Color.Lerp(activeColor, restColor, normalizedEelapsedTime);
		for(metronomeLED in metronomeLEDs) {
			SetMetronomeLEDColor(metronomeLED, lerpedColor);
		}
		yield;
	}
	
	for(metronomeLED in metronomeLEDs) {
		SetMetronomeLEDColor(metronomeLED, restColor);
	}
}

File: Assets/Scripts/OSC/OSCMetronomeLED.cs
using UnityEngine;
using System.Collections;

public class OSCMetronomeLED : MonoBehaviour 
{
	public int channel = 1;
	public Color restColor = Color.black;
	public Color activeColor = Color.red;
	public MetronomeLED[] metronomeLEDs;
	
	[System.Serializable]
	public class MetronomeLED {
		public GameObject obj;
		
		[System.NonSerialized] 
		public Material mat;
	}
	
	void Start () {
		
		// initialize all LEDs
		foreach(MetronomeLED metronomeLED in metronomeLEDs) {
			metronomeLED.mat = metronomeLED.obj.gameObject.GetComponent<Renderer>().material;
			SetMetronomeLEDColor(metronomeLED, restColor);
		}
	}
	
	void SetMetronomeLEDColor(MetronomeLED metronomeLED, Color inputColor) {
		metronomeLED.mat.SetColor("_Color", inputColor);
		metronomeLED.mat.SetColor("_TintColor", inputColor);
		metronomeLED.mat.SetColor("_Emission", inputColor);
		//metronomeLED.mat.SetColor("_TintColor", inputColor);
	}
	
	void OSCMetronome (float tempo) {
		StopCoroutine( "StrobeLED");	
		StartCoroutine( "StrobeLED", 0.5);
	}
	
	IEnumerator StrobeLED(float fadeTime) {
		float startTime = Time.time;
		float normalizedEelapsedTime = 0.0f;
		
		while(normalizedEelapsedTime <=1.0){
			float elapsedTime = (Time.time - startTime);
			normalizedEelapsedTime = elapsedTime/fadeTime;
			Color lerpedColor = Color.Lerp(activeColor, restColor, normalizedEelapsedTime);
			foreach(MetronomeLED metronomeLED in metronomeLEDs) {
				SetMetronomeLEDColor(metronomeLED, lerpedColor);
			}
			yield return null;
		}
		
		foreach(MetronomeLED metronomeLED in metronomeLEDs) {
			SetMetronomeLEDColor(metronomeLED, restColor);
		}
	}
}

